packages/devtools_shared/lib/src/directory_picker/directory_picker.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:file/local.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+import 'scripts.dart';
+
+class DirectoryPickerManager {
+  /// The key to retrieve error message from the returning map of this class's
+  /// APIs.
+  static const kErrorField = 'error';
+
+  /// The key to retrieve output json from the returning map of this class's
+  /// APIs.
+  static const kOutputTextField = 'text';
+
+  String get scriptRoot => path.dirname(Platform.script.path);
+
+  Future<String> launchDirectoryPicker() {
+    if (Platform.isMacOS) {
+      return _launchDirectoryPickerMacOS();
+    }
+    throw UnimplementedError();
+  }
+
+  Future<String> _launchDirectoryPickerMacOS() async {
+    // Write script into a temp file.
+    final tempDirectory =
```

Comment:
unfortunately, I can't find a way to get the path of the current package, so I instead write the script into a temp file and run it.